### PR TITLE
fix(globalaccelerator-endpoints): add preserveClientIp option for net…

### PR DESCRIPTION
**Pull request by sj5n**
…work loadbalancer

### Reason for this change

preserveClientIp was missing for GlobalAccelerator Endpoints when using a network loadbalancer.

### Description of changes

* add missing network load balancer endpoint prop.

### Description of how you validated changes

Added unit tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

### CODE REVIEW

**Comment by SaschaJanssen** [comment]
Exemption Request: No integration but unit tests added. fix just contains a new property to the global accelerator endpoint.

File: packages/aws-cdk-lib/aws-globalaccelerator-endpoints/lib/nlb.ts [hunk 1]
```diff
@@ -14,6 +14,19 @@ export interface NetworkLoadBalancerEndpointProps {
...
+  /**
+   * Forward the client IP address in an `X-Forwarded-For` header
+   *
+   * GlobalAccelerator will create Network Interfaces in your VPC in order
+   * to preserve the client IP address.
+   *
+   * Client IP address preservation is supported only in specific AWS Regions.
+   * See the GlobalAccelerator Developer Guide for a list.
+   *
+   * @default true if available
```
Comment (scorbiere) [inline packages/aws-cdk-lib/aws-globalaccelerator-endpoints/lib/nlb.ts:27]: According to the documentation, my understanding is that the default value is `false` : 
https://docs.aws.amazon.com/global-accelerator/latest/dg/preserve-client-ip-address.html
> When you add an Network Load Balancer with security groups as an endpoint in Global Accelerator, client IP address preservation is not enabled by default.

```suggestion
   * @default false
```

**Review by scorbiere — CHANGES_REQUESTED** [review]
Thank you for your contribution. Would you mind updating the default value in the comment? Then I can approve the PR.

**Comment by SaschaJanssen** [comment]
@scorbiere thanks, updated the doc.

**Review by scorbiere — APPROVED** [review]
LGTM, thanks for the update.

**Comment by scorbiere** [comment]
@mergify update

**Review by scorbiere — APPROVED** [review]
Re-approving after updating the branch.

**Pull request merged**